packages/number-field/src/vaadin-number-field-mixin.js

```diff
@@ -383,6 +393,113 @@ export const NumberFieldMixin = (superClass) =>
      */
     _setHasInputValue(event) {
       const target = event.composedPath()[0];
-      this._hasInputValue = target.value.length > 0 || target.validity.badInput;
+      this._hasInputValue = target.value.length > 0 || this.__hasUnparsableValue;
+    }
+
+    /**
+     * Override this method from `InputMixin` to prevent
+     * the value change caused by user input from being treated
+     * as initiated programmatically by the developer and therefore
+     * from getting silently committed by the value observer without
+     * any change event. The value change will be committed later
+     * on blur or Enter.
+     *
+     * @param {InputEvent} event
+     * @override
+     * @protected
+     */
+    _onInput(event) {
```

Comment:
This method is marked as `@private` in the `InputMixin`. Should it be changed to `@protected`?

Reply:
It seems there may have been some confusion. There are actually two methods in `InputMixin`, differing only by the underscore: `_onInput` and `__onInput`.

I'm overriding `_onInput`, and as far as I can see, this one is marked as protected in `InputMixin`

Reply:
Ah, you are right. Sorry my bad. The `__` got me there. Never mind, moving on 😅